src/adk/api/local_api.py

```diff
@@ -371,47 +371,41 @@ def delete_application(self, application_name: Optional[str], path: Path) -> boo
         """
         application_dir_deleted = False
         all_subdir_deleted = True
-        application_path = path / application_name if application_name is not None else path
-        # When we are not in the directory of the application, get the directory from the config
-        if not application_path.is_dir() and application_name is not None:
-            app_path = self.__config_manager.get_application_path(application_name)
-            if app_path is not None:
-                application_path = Path(app_path)
 
         # check if the application exists and save the app name to delete it later
-        application_name_from_config, _ = self.__config_manager.get_application_from_path(application_path)
+        application_name_from_config, _ = self.__config_manager.get_application_from_path(path)
 
         # Check if application path exists
-        if application_path.is_dir():
-            application_src_path = application_path / 'src'
+        if path.is_dir():
```

Comment:
I would propose to consistently use application_path for the complete path to the application as stored in config and returned by retrieve_application_name_and_path. The same for experiment_path. Now you changed it the other way around here.
Thus:

application_path = D:\dev\qne-adk\applications\application1
application_name = application1
path = D:\dev\qne-adk\applications

In application_create and __create_application_structure we use path because that is the path to create the application in. Only after creation of this application we have an application_path (as the variable is named correctly in __create_application_structure ).